Buttplug.Server/Bluetooth/Devices/FleshlightLaunch.cs

```diff
@@ -75,18 +79,63 @@ private Task<ButtplugMessage> HandleStopDeviceCmd(ButtplugDeviceMessage aMsg)
             return Task.FromResult<ButtplugMessage>(new Ok(aMsg.Id));
         }
 
+        // Speed returns the speed (in percent) to move the given distance (in percent)
+        // in the given duration.
+        // Thanks to @funjack - https://github.com/funjack/launchcontrol/blob/master/protocol/funscript/functions.go
+        private double GetSpeed(double aDistance, uint aDuration)
+        {
+            if (aDistance <= 0)
+            {
+                return 0;
+            }
+            else if (aDistance > 1)
+            {
+                aDistance = 1;
+            }
+
+            double mil = Convert.ToDouble(aDuration) * 90 * aDistance;
+            return Math.Min(250 * Math.Pow(mil, -1.05), 1);
+        }
+
         private async Task<ButtplugMessage> HandleFleshlightLaunchRawCmd(ButtplugDeviceMessage aMsg)
         {
             // TODO: Split into Command message and Control message? (Issue #17)
-            var cmdMsg = aMsg as FleshlightLaunchFW12Cmd;
-            if (cmdMsg is null)
+            var cmdMsg1 = aMsg as FleshlightLaunchFW12Cmd;
+            var cmdMsg2 = aMsg as LinearCmd;

```

Comment:
Please make a new method for handling LinearCmd, do what you need to for Linear there. Combining all of this into the same handler doesn't make sense.

Reply:
It was a pattern that started in the vibration devices, where most of the logic was the same: duplicating the same logic twice seemed wasteful.

I'll refactor so that they become multiple handler methods that chain rather one that internally switches.

Reply:
You can create a function both message handlers call in order to not repeat code. I want it to be clear what each device does with each message, which is why I created the message sorter in the functional manner I did.